social/backends/open_id.py

```diff
@@ -360,3 +437,16 @@ def request_access_token(self, *args, **kwargs):
         response = self.get_json(*args, **kwargs)
         self.id_token = self.validate_and_return_id_token(response['id_token'])
         return response
+
+    def user_data(self, access_token, *args, **kwargs):
+        return self.get_json(self.USERINFO_URL,
+                             headers={'Authorization': 'Bearer {0}'.format(access_token)})
+
+    def get_user_details(self, response):
+        return {
+            'username': response.get('preferred_username'),
+            'email': response.get('email'),
+            'fullname': response.get('name'),
+            'first_name': response.get('given_name'),
+            'last_name': response.get('family_name'),
+        }
```

Comment:
I was mostly talking about `username_key = self.setting('USERNAME_KEY') or self.USERNAME_KEY`

Which lets users override the username in their own settings file and exists in other backends.